xsimlab/xr_accessor.py

```diff
@@ -500,42 +503,44 @@ def update_clocks(self, model=None, clocks=None, master_clock=None):
 
         ds = self._ds.copy()
 
-        if isinstance(master_clock, str):
-            master_clock_dict = {"dim": master_clock}
+        if master_clock is not None:
```

Comment:
```suggestion
        if master_clock is not None and main_clock is None:
```

In the (rare) case where the user provides a value for both arguments, I think we can safely ignore the value given for `master_clock`.